### PR TITLE
splitting processing into separate ephys and imaging processing  schemas

### PR DESCRIPTION
**Pull request by saskiad**
still work in progress

### CODE REVIEW

File: schemas/ephys/ephys_processing.json [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                "notes": {
+                    "type": "string"
+                },
+                "name": {
+                    "anyOf": [
+                        {
+                            "title": "preprocessing",
+                            "type": "object",
+                            "properties": {
+                                "highpass_filter": {
```
Comment (dyf) [inline schemas/ephys/ephys_processing.json:82]: do we ever lowpass or bandpass?  wonder if we should generalize this.

File: schemas/ephys/ephys_processing.json [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                                            "enum": ["median"]
+                                        }
+                                    },
+                                    "required": ["reference", "operator"]
+                                }
+                            },
+                            "required": ["highpass_filter", "phase_shift", "common_reference"]
+                        },
+                        {
+                            "title": "align_params_Ex_488_Em_525",
```
Comment (dyf) [inline schemas/ephys/ephys_processing.json:128]: is this supposed to be in imaging instead of ephys?

File: schemas/ephys/ephys_processing.json [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                            "type": "object",
+                            "properties": {
+                                "cpu_params": {
```
Comment (dyf) [inline schemas/ephys/ephys_processing.json:131]: These names are all super confusing. I don't have any idea what they could mean.

File: schemas/imaging/imaging_processing.json [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+                                    "type": "integer",
+                                    "title": "Channel excitation (nm)",
+                                    "units": "nm"
+                                },
+                                "channel_emission": {
+                                    "type": "integer",
+                                    "title": "Channel emission (nm)",
+                                    "units": "nm"
+                                },
+                                "reference_1": {
```
Comment (dyf) [inline schemas/imaging/imaging_processing.json:133]: I assume this is what terastitcher wants, but I'm wondering if we can reuse the coordinate array format from the other schemas.

File: schemas/imaging/imaging_processing.json [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+                                "voxel_size_1",
+                                "reference_2",
+                                "voxel_size_2",
+                                "reference_3",
+                                "voxel_size_3"
+                            ]
+                        },
+                        {
+                            "title": "import_params_multivolume",
+                            "type": "obejct",
```
Comment (dyf) [inline schemas/imaging/imaging_processing.json:184]: typo

File: schemas/imaging/imaging_processing.json [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+                    "description": "Path to code respository"
+                },
+                "notes": {
+                    "type": "string"
+                },
+                "name": {
+                    "anyOf": [
+                        {
+                            "title": "import",
+                            "type": "obejct",
```
Comment (dyf) [inline schemas/imaging/imaging_processing.json:121]: typo

File: schemas/imaging/imaging_processing.json [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+                            "title": "merge_volume",
+                            "type": "object",
+                            "properties": {
+                                "parallel_params": {
+                                    "type": "array",
+                                    "items": {"$ref": "#/$defs/parallel"},
+                                    "uniqueItems": true,
+                                    "title": "Parallel parameters"
+                                },
+                                "s": {
```
Comment (dyf) [inline schemas/imaging/imaging_processing.json:289]: yikes

**Review by dyf — COMMENTED** [review]
A couple top level comments:

- I think it's still good to have a generic and simple processing schema at the top level
- I see now uninterpretable the module parameters are right now.  Could go far by a) giving the a clearer title (e.g. what is `merge_volume`?), and b) writing better descriptions.  But! It will be hard to keep this in sync with unstable code.  We need to be careful about this.
- There's some repetition, but I'm sure you are aware of this.

File: schemas/ephys/ephys_processing.json [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+                    "title": "Code URL",
+                    "description": "Path to code respository"
+                },
+                "notes": {
+                    "type": "string"
+                },
+                "name": {
+                    "anyOf": [
+                        {
+                            "title": "preprocessing",
```
[the remark this answers is not in the document]
Comment (saskiad) [reply schemas/ephys/ephys_processing.json:79]: isn't that what these are?

File: schemas/imaging/imaging_processing.json (hunk 4 above)
Comment (saskiad) [reply schemas/imaging/imaging_processing.json:133]: it's too different from the coordinate array we have for ephys, but I do think I can make a subschema for this specifically...

File: schemas/ephys/ephys_processing.json (hunk 2 above)
Comment (saskiad) [reply schemas/ephys/ephys_processing.json:128]: no, I just haven't deleted yet from ephys... I duplicated the same file into two and was editing one and just had the other as a place holder.

File: schemas/imaging/imaging_processing.json (hunk 7 above)
Comment (saskiad) [reply schemas/imaging/imaging_processing.json:289]: yes, Camillo doesn't know what they are ... so right now we're stuck with them

**Comment by saskiad** [comment]
@dyf check out this generic processing.json

**Comment by dyf** [comment]
I like the new generic schema.

**Pull request merged**